src/ray/protobuf/autoscaler.proto

```diff
@@ -0,0 +1,160 @@
+syntax = "proto3";
+option cc_enable_arenas = true;
+package ray.rpc;
+
+message ResourceBundle {
+  // The fixed point resource size of the resource (granularity: 1/10000).
+  map<string, int64> resources = 1;
+}
+
+// A workload (task/actor/placement group) unfulfilled by the the cluster's internal scheduler(s).
+message QueuedResources {
+
+  // Type of placement group strategy.
+  enum PlacementStrategy {
+    // Packs Bundles into as few nodes as possible.
+    PACK = 0;
+
+    // Places Bundles across distinct nodes or processes as even as possible.
+    SPREAD = 1;
+
+    // Packs Bundles within one node. The group is not allowed to span multiple nodes.
+    STRICT_PACK = 2;
+
+    // Places Bundles across distinct nodes.
+    // The group is not allowed to deploy more than one bundle on a node.
+    STRICT_SPREAD = 3;
+  }
+
+  // The number of resources of this quantity. This should be
+  // functionally equivalent to setting quantity=1 and repeating the
+  // field many times.
+  uint64 quantity = 1;
+
+  // How badly the cluster needs the resource 0-20. 20 means the
+  // cluster cannot complete the workload without satisfying the
+  // resource (e.g. a "hard" demand). 0 means the resource is only
+  // needed for convenience.
+  int32 necessity = 2;
+
+  // Each individual resource bundle in the queued resources. For
+  // tasks or actors this is length 1.
+  repeated ResourceBundle bundles = 3;
+
+  // The placement strategy of bundles within this queued resources.
+  PlacementStrategy placement_strategy = 4;
+}
+
+message GetAllQueuedResourcesRequest {
+}
+
+message GetAllQueuedResourcesResponse {
+  // The set of all queued resources.
+  repeated QueuedResources queued_resources = 1;
+}
+
+message NodeInfo {
+  // State of a node.
+  enum NodeState {
+    // The node has not joined the cluster yet.
+    PENDING = 0;
+
+    // The node has joined the cluser and is healthy.
+    ALIVE = 1;
+
+    // The node is marked for removal, but it may not be safe to remove it yet.
+    DRAINING = 2;
+
+    // The node is no longer active in the cluster it is safe to remove it.
+    DEAD = 3;
+  }
+
+  // The state of the node.
+  NodeState node_state = 1;
+
+  // TODO: Ideally we unify GCS's node id with the autoscaler's here.
+  // The ID of node.
+  bytes node_id = 2;
+
+  // The address of the node manager service for the node.
+  int32 node_manager_port = 3;
+  string node_manager_address = 4;
+
+  // The total fixed point resources (granularity 1/10000) of this node.
```

Comment:
change "total" to "available"

Reply:
assuming this comment was intended for the comment over `resources_available` which I updated.